src/config.cc

```diff
@@ -2235,12 +2233,8 @@ void save( Class const & c )
 
   QByteArray result( dd.toByteArray() );
 
-  if ( configFile.write( result ) != result.size() )
+  if ( configFile.write( result ) != result.size() || configFile.commit() )
```

Comment:
configFile.commit()  or  ! configFile.commit()

Reply:
if `configFile.write( result ) != result.size()` evaluate to true ,the configFile.commit() will get no chance to run.
from the doc of QSaveFile ,the commit method should be called mandatory

Reply:
`!` 

:)

Reply:
No, commit() is mandatory if we want to copy the tmp file to target.

`configFile.write( result ) != result.size()` hints something wrong, thus no need to commit().

Reply:
![image](https://github.com/xiaoyifang/goldendict-ng/assets/105986/dbdbc4ff-16ed-48d2-8194-aa4a94e1b0da)

the commit method should be handling the write failure check .


Can this line 
`if ( configFile.write( result ) != result.size() || !configFile.commit() )`
reduce to 
`if(!configFile.commit())` alone.

Reply:
sounds ok.

Reply:
Actually, I think the `configFile.write( result ) != result.size()` is not needed.

QIODevice internally keeps a errorString, and QSaveFile interanlly have another writeError,

The commit() will check both,

So we don't need to manually check

Reply:
configFile.write( result ) is still needed.

 configFile.write( result );
if(configFile.commit()){

Reply:
Fixed,

The rabbit hole:

QFile's write error check https://github.com/qt/qtbase/blob/ae8031b5e72032f9e2884c18cd72639acfd0d1a4/src/corelib/io/qsavefile.cpp#L307

rename failure check https://github.com/qt/qtbase/blob/ae8031b5e72032f9e2884c18cd72639acfd0d1a4/src/corelib/io/qsavefile.cpp#L315

The write error is done by reimplement QIODevice's writeData https://github.com/qt/qtbase/blob/ae8031b5e72032f9e2884c18cd72639acfd0d1a4/src/corelib/io/qsavefile.cpp#L362-L363